### ISSUE TITLE
[BUG] guess estimate user behind the obfuscation 

### ISSUE THREAD

**Issue by slashdd** [soscleaner/soscleaner#129]
I notice a possible pattern (reproducible in all my Ubuntu test so far) where root always seems to refer to obfuscateduser1.

I haven't investigate enough (will do when I'll have time) to tell if it's lastlog which always place root at the top of its list (or maybe a pure coincidence ?) or the way soscleaner build the user list (most likely following the order of lastlog)

Might be a good idea to shuffle the user before the obfuscation process (and maybe more data to obfuscated if needed) found in lastlog (or other source of obfuscation type), so that one can't predict/guess estimate the data behind the obfuscate data name.

Shuffling will help but I also agree that finding the root user from its obfuscated counterpart is fairly easy just by looking the process table inside 'ps' and look for processes that normally run as root, so I don't think there is a clean way to avoid any suspicion, but I think shuffling may be a good start, at least to prevent root to always refer to obfuscateduser1.

Does this pattern also reproducible inside RHEL ?

Example:
obfuscateduser1         1  0.0  0.0 225084  9128 ?        Ss   Jul02   0:01 /sbin/init

I don't know if this can be considered a security issue, but I would definitely consider this bug with high importance since the goal of soscleaner is to not reveal sensible information such as user, and for now root can easily be predicted.

**Comment by slashdd** [comment]
This could potentially be the culprit here (tbd)
```
    def _prime_userdb(self):
        """Creates an initial entry in the user_db for the root user. This is needed so we
        have something to iterate through for later functions.
        """

        try:
            new_user = "obfuscateduser%s" % self.user_count
            self.user_db['root'] = new_user

            return True
```

**Comment by slashdd** [comment]
The above piece of code is indeed the offending one. 
It is always placing root on top of the list, thus always defaulting to obfuscateduser1.

**Comment by slashdd** [comment]
1) root should not always be the first in the list
2) obfuscateduser count doesn't need to be in a numeric order, the number can also be shuffled.
3) username in lastlog can also be shuffled.

These are ideas I have at the moment to reduce the risk of prediction.

**Comment by jduncan-rva** [comment]
Fixed with cf657c5
